samples/hosting/docker/Core_7/Receiver/Program.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using NServiceBus;
+using NServiceBus.Logging;
+
+class Program
+{
+    static async Task Main(string[] args)
+    {
+        //required to prevent possible occurrence of .NET Core issue https://github.com/dotnet/coreclr/issues/12668
+        Thread.CurrentThread.CurrentUICulture = new CultureInfo("en-US");
+        Thread.CurrentThread.CurrentCulture = new CultureInfo("en-US");
+
+        Console.CancelKeyPress += new ConsoleCancelEventHandler(OnExit);
+
+        Console.Title = "Samples.Docker.Receiver";
+        LogManager.Use<DefaultFactory>()
+            .Level(LogLevel.Info);
+
+        var endpointInstance = await TryStart()
+            .ConfigureAwait(false);
+        
+        Console.WriteLine("Press Ctrl+C to exit.");
+
+        // Wait until the message arrives.
+        closingEvent.WaitOne();
+
+        await endpointInstance.Stop()
+            .ConfigureAwait(false);
+    }
+
+    static async Task<IEndpointInstance> TryStart()
+    {
+        int count = 0;
+
+        while (count < 5)
```

Comment:
what is the purpose of multiple start attempts?

Reply:
Compose starts the RabbitMQ container first but there's a delay between when the container is up and when the broker can actually accept connections. First attempt to connect always fails, second usually is OK.

Reply:
Would it make sense to have a comment in the code explaining this?

Reply:
also lets make this explicit


    static async Task WaitForRabbitToStart()
    {
        var factory = new ConnectionFactory
        {
            Uri = "amqp://localhost"
        };
        for (var i = 0; i < 5; i++)
        {
            try
            {
                using (factory.CreateConnection())
                {
                }
                return;
            }
            catch (BrokerUnreachableException)
            {
            }
            await Task.Delay(1000).ConfigureAwait(false);
        }
    }

Reply:
@Particular/rabbitmq-transport-maintainers BTW i tried to do the above with a connection timeout of 5 seconds, but it seemed to ignore that timeout. have u ever got that working?

Reply:
Thanks, I will incorporate this.